airbyte/caches/motherduck.py

```diff
@@ -0,0 +1,31 @@
+"""A cache implementation for the MotherDuck service, built on DuckDB."""
+from __future__ import annotations
+
+from overrides import overrides
+
+from airbyte.caches.duckdb import DuckDBCacheBase, DuckDBCacheConfig
+
+
+class MotherDuckCacheConfig(DuckDBCacheConfig):
+    """Configuration for the MotherDuck cache."""
+
+    db_path = "md:"
+    database: str
+    api_key: str
+
+    @overrides
+    def get_sql_alchemy_url(self) -> str:
+        """Return the SQLAlchemy URL to use."""
+        # return f"duckdb:///{self.db_path}?schema={self.schema_name}"
+        return f"duckdb:///md:{self.api_key}@{self.database}"
```

Comment:
From Jordan [in Slack](https://airbytehq-team.slack.com/archives/C05KJKWHNAJ/p1708127088668119?thread_ts=1708042440.842169&cid=C05KJKWHNAJ):

> One thing, from the PR, it looks like you're using an old version of the sqlalchemy connection...legacy reasons).
> ...
> It would be better to switch to this syntax:
> f:"duckdb:///md:{self.database}?motherduck_token={self.api_key}"
> Note that this is closer to the existing duckdb connection string
> ...
> and...be able to use schema name too with:
> f"duckdb:///md:{self.database}?schema={self.schema_name}&motherduck_token=<my_token>"

Reply:
Updated! ✅